### PR TITLE
Added "cullingMask" feature to Camera. (updated)

### PR DESCRIPTION
**Pull request by aidinabedi**
**Update:**
This is an updated pull request based on comments and feedback from the [old an (removed) pull request](https://github.com/playcanvas/engine/pull/674).

**Description:**
Include or omit which objects are to be rendered by a Camera.

**Motiviation:**
Currently there is no way to control which objects are rendered by a camera. This is a significant limitation, especially when rendering to texture, as you typically want to hide some objects that may be an obstruction from another camera angle or interfere with render target results. Another example is to have multiple cameras in the same scene that render different objects, and combine the results in post effects (such as outlining in the PlayCanvas Editor). In addition, this feature has also been requested in the [forums](http://forum.playcanvas.com/t/camera-mask-to-control-which-instance-to-render/940).

**API changes:**
- New property `Camera.cullingMask`.

**Implementation:**
The implementation is quite simple and reuses the existing mask property [`MeshInstance.mask`](http://developer.playcanvas.com/en/api/pc.MeshInstance.html#mask). If the object's mask AND the camera's cullingMask is zero then the game object will be invisible from the camera. To achieve this feature only minor changes where necessary in the scene's internal culling method. The new `Camera.cullingMask` property is inspired by familiar APIs such as [Unity's Camera.cullingMask](https://docs.unity3d.com/ScriptReference/Camera-cullingMask.html).

### CODE REVIEW

File: src/scene/forward-renderer.js [hunk 1]
```diff
@@ -974,11 +974,17 @@ pc.extend(pc, function () {
...
+            var cullingMask = camera.cullingMask || 0xFFFFFFFF; // if missing assume camera's default value
+
             if (!camera.frustumCulling) {
                 for (i = 0; i < drawCallsCount; i++) {
                     // need to copy array anyway because sorting will happen and it'll break original draw call order assumption
                     drawCall = drawCalls[i];
                     if (!drawCall.visible && !drawCall.command) continue;
+
+                    // if the object's mask AND the camera's cullingMask is zero then the game object will be invisible from the camera
+                    if (drawCall.mask && ((1 << drawCall.mask) & cullingMask) === 0) continue;
```
Comment (guycalledfrank) [inline src/scene/forward-renderer.js]: You don't actually need "1 <<" here, because drawCall.mask is already a bitmask ready to be AND'ed

**Comment by Maksims** [comment]
Could you please sign this [Contributors Agreement Form](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform), so we can merge your PR?
Thanks a lot for making a PR :)

File: src/scene/forward-renderer.js (hunk 1 above)
Comment (aidinabedi) [inline src/scene/forward-renderer.js]: Of course, "1 <<" should be removed!

**Comment by aidinabedi** [comment]
Signed!

File: src/scene/forward-renderer.js (hunk 1 above)
Comment (aidinabedi) [inline src/scene/forward-renderer.js]: Fix it in commit below

**Comment by guycalledfrank** [comment]
Thanks! Seems to work.

**Pull request merged**